core/src/driver/stablex_driver.rs

```diff
@@ -127,16 +154,26 @@ impl<'a> StableXDriverImpl<'a> {
         };
 
         let submitted = if let Some(objective_value) = verified {
+            if let Ok(duration) = (batch_to_solve.solve_start_time() + min_solution_submit_time)
+                .duration_since(SystemTime::now())
+            {
+                self.sleep.sleep(duration).await;
```

Comment:
I know Nick suggested removing the sleep from SolutionSubmission and instead I've added another sleep here.
I attempted that refactoring but it is more controversial and complex so I want this PR to focus on this change and overall. I am not sure whether that refactoring makes sense and it's not needed now.